pkg/adaptertesting/adapters.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package adaptertesting
+
+import (
+	"testing"
+
+	"istio.io/mixer/pkg/aspect"
+	"istio.io/mixer/pkg/aspect/denyChecker"
+	"istio.io/mixer/pkg/aspect/listChecker"
+	"istio.io/mixer/pkg/registry"
+)
+
+// RegisterFunc is the function that registers adapters into the supplied registry
+type RegisterFunc func(registry.Registrar) error
+
+type fakeRegistrar struct {
+	registrations int
+}
+
+func (r *fakeRegistrar) RegisterCheckList(listChecker.Adapter) error {
+	r.registrations++
+	return nil
+}
+
+func (r *fakeRegistrar) RegisterDeny(denyChecker.Adapter) error {
+	r.registrations++
+	return nil
+}
+
+// TestAdapterInvariants ensures that adapters implement expected semantics.
+func TestAdapterInvariants(a aspect.Adapter, r RegisterFunc, t *testing.T) {
+	if a.Name() == "" {
+		t.Error("All adapters need names")
```

Comment:
For these tests, can we start the error log line with something indicating the method and/or adapter?

Maybe something like:
"Name() => "", all adapters need names"
"Description() => "", all adapters need descriptions"
"ValidateConfig(%T) => unexpected error for DefaultConfig(): %v", c, err
"Close() => unexpected error: %v", err
"RegisterFunc() => unexpected error: %v", err
"RegisterFunc() => %d registrations, want at least 1", fr.registrations
?